duecredit/tests/test_io.py

```diff
@@ -211,3 +212,10 @@ def test_enumeratedentries():
     assert_raises(KeyError, enumentries.fromrefnr, 666)
 
     assert_equal(entries, sorted(enumentries, key=lambda x: x[1]))
+
+def test_get_text_rendering():
+    citation_bibtex = Citation(_sample_bibtex, path='mypath')
+    citation_doi = Citation(_sample_doi, path='mypath')
+    # smoke test
+    assert_true(get_text_rendering(citation_bibtex))
```

Comment:
On Sat, 15 Aug 2015, Matteo Visconti di Oleggio Castello wrote:
>    this won't work. hints?

because there is a bug! ;)  aren't tests great? ;)  although bug is only in the
test itself.

I first thought that it is due to some intricacies of mock, so you might
like to read
http://alexmarandon.com/articles/python_mock_gotchas/
http://www.voidspace.org.uk/python/mock/patch.html#where-to-patch

but then, after I added   

import pdb; pdb.set_trace()

right before

assert_true(get_text_rendering(citation_bibtex))

it pointed that citation.entry is not BibTeX, it is a str!  So we have a bit
"ambigous" API here for the Citation -- entry could be a string key or an
actual DueCreditEntry.  Would be nice to add some safeguard I guess, e.g. throw
a warning if provided entry is a string and has new lines in it (unlikely we
should allow keys to have new lines).  Or may be  we should make it more
explicit??? (make it have two separate parameters -- entry and key)

anyways -- fix here would be 

citation_bibtex = Citation(BibTeX(_sample_bibtex), path='mypath')

and then failure would be different and more meaningful ;)